scripts/scripts.js

```diff
@@ -57,11 +60,25 @@ const miloLibs = setLibs(LIBS);
   });
 }());
 
+const setUserProfile = async () => {
+  let profileData = getLocalStorage('profile');
+  if (profileData === null) {
+    const response = await fetch(PROFILE_URL, {
+      method: 'GET',
+      headers: { 'Content-Type': 'application/json' },
+    });
+    profileData = await response.json();
+  }
+  const details = profileData?.details;
```

Comment:
Thought :: Right now I could see that with sessionStorage, a n/w call is being done for every new session to fetch user profile details. Though the API endpoint should be able to handle the request load right now; you may consider moving to an "expiry-based" localStorage to optimise it further; the need for that arises.

You could go through one approach here - https://www.sohamkamani.com/javascript/localstorage-with-ttl-expiry/